### DISCUSSION TITLE
Disable battery status upload?

### DISCUSSION THREAD

**Discussion by shafiro** [NightscoutFoundation/xDrip#2681] [General]
Hi.
We use Xdrip as a CGM on the master device, and on one of the following devices as an uploader of BG to NS and TP.
NS gets the battery status (%) of those 2 devices, showing the lower number (%) on top, and since we are only interested in the master battery status, this causes confusion.
I could not find a way to filter the follower device battery status from showing on NS.
Is there a way in Xdrip to disable the uploading of the battery status for a specific device?

**Comment by Navid200** [comment]
What do you see on the following page of xDrip that uploads to Nightscout?

Settings &#8722;> Cloud Upload &#8722;> Nightscout Sync &#8722;> Extra Options

**Reply by shafiro** [reply]
Hi. Thank you.
None of the 6 options are enabled.

**Reply by Navid200** [reply]
If you are talking about the battery of the phone that is uploading to Nightsout, I don't think there is a way to disable that.

Is there @psonnera?

**Reply by shafiro** [reply]
Thank you.
Yes, about the phone battery.
I thought it is better not to upload irrelevant data than trying to filter it in NS.

**Comment by Navid200** [comment]
I'm sorry for coming back to this discussions after such a long time.  But, I would like to have this here because I feel it is directly related to what has been requested by this discussion.

Is there a reason the master cannot also upload to Nightscout?
That would address the issue, right?

**Reply by psonnera** [reply]
To me, master phone should upload its battery status to Nightscout. This is an important feature when you're following kids. As for followers, I don't see the value.

**Comment by shafiro** [comment]
I cannot see the point in uploading battery status of devices other than the master. Howevev I may be missing a certain situation, so I think it should at least be possible to filter/exclude this data.

**Reply by Navid200** [reply]
You use a follower scheme to get readings from master.  Then, you use the follower phone to upload to Nightscout.
Nightscout uploader uploads the battery level of the uploader phone because it was assumed that the uploader phone is the master phone.

You are not using the master phone to upload to Nightscout.  That has caused an unexpected behavior.
One solution is to complicate the uploader program as you suggest and add a filter.
An easier solution is to use the master phone to upload to Nightscout.

Before we go ahead and complicate the xDrip program, I need to be sure there is a reason you need to do what you are doing.
So, I am going to ask the question again:

Is there a reason the master does not also upload to Nightscout?

**Comment by shafiro** [comment]
Yes. Any task that can be offloaded from the master saves battery life so I use a follower for the uploads (NS and Tidepool in my case). There is no need for a filter. Since followers battery level is irrelevant, it should not be uploaded. I actually see it as simplifying the xDrip program.

**Reply by Navid200** [reply]
It will be great if you can open a PR and show us how you can accomplish this by also simplifying the xDrip code.
The way I see it, if I was to do it, I would have to add a conditional to check if the phone is a follower or not and make the upload of the battery conditional on that.
But, the problem is that if xDrip is used correctly, you would only upload data to Nightscout from the master.  

If you have a master and intend to use Nightscout, why don't you set it up to upload to Nightscout only and set up the followers to follow Nightscout?  

You don't have to answer my questions.  But, unless there is a logic for a change to xDrip, I will disagree with it.  xDrip is already too large and complex.  There is no reason to add to it unless absolutely necessary.

**Comment by shafiro** [comment]
I'm not a developer but a user, I apologize if that wasn't clear. My gratitude to the xDrip developers and to you personally (you've helped me several times) is undeniable. I probably misunderstood the reason you raised this issue after a year of inactivity. Anyway, I've been using a follower to upload BG data for years without any problems, it saves some battery life on my teenage son's master device, who insists on charging his phone only after it's completely turned off. For the record, I don't know how many lines of code are required to add a conditional to check if the phone is a follower or not and make the upload of the battery conditional on that. But could it possibly be shorter and more efficient to simply enable battery upload when the "act as a master" box is checked?

**Reply by Navid200** [reply]
There was a request a day or two ago to add the ability to download a master battery level from Nightscout by a follower.
I had to do a search to se what else has come up that may be relevant and found this thread.  It seemed like an incomplete conversation and I wanted to get it to a conclusion.  

xDrip is very complicated because it doesn't just do one thing.
If you want to use Dexcom app for G5, you download an app just for that.  You download a completely different app if you want to use Dexcom app for G6.  Yet, you download another for G7.
Now, if you want to use a Dexcom app to follow someone, yet, you have to download another app.  Every one of these Dexcom apps does only one thing and there is a company with full-time employees behind them all.
xDrip does all of that in a single app.   But, it can also upload to Nightscout not just xDrip Sync.  But, it can also upload to Tidepool.  It can also get readings from Libre.
It can also get readings from a companion app.

All of this is there at a cost, the complexity of the code.
Every time we have to update xDrip for a new Android level, we have to fix lots of incompatibilities.  This means that adding a new feature is not the end of a challenge.  In fact, it could be considered just the beginning!  A lot of time goes into maintaining xDrip.
I am sorry for all of this detail.  
I use xDrip and I would like us to be able to use it for a very long time.  We have to be very careful in how we add more and more and more features to it.  It could lead to complications.  
I'm not against improvements.  I just would like to make sure a change is absolutely necessary before we make it.

Thanks for the conversation.
